dbms/src/Flash/Disaggregated/GRPCPageReceiverContext.h

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/UnaryCallback.h>
+#include <Common/grpcpp.h>
+#include <Flash/Coprocessor/ChunkCodec.h>
+#include <Flash/Mpp/MPPTaskManager.h>
+#include <Storages/DeltaMerge/Remote/RemoteReadTask.h>
+#include <Storages/StorageDisaggregated.h>
+#include <common/types.h>
+#include <grpcpp/completion_queue.h>
+#include <kvproto/mpp.pb.h>
+#include <pingcap/kv/Cluster.h>
+#include <tipb/executor.pb.h>
+
+#include <memory>
+
+namespace DB
+{
+using PageDataPacket = mpp::PagesPacket;
+// TODO make the memory tracked
+using TrackedPageDataPacketPtr = std::shared_ptr<PageDataPacket>;
+using TrackedPageDataPacketPtrs = std::vector<TrackedPageDataPacketPtr>;
+
+class ExchangePagePacketReader
```

Comment:
The name `exchange` is ambiguous (with Exchange executor). How about calling it something like `FetchPagesStreamReader`?

Reply:
Yes, it is a typo.